lib/kubeclient/http_proxy_wrapper.rb

```diff
@@ -0,0 +1,49 @@
+module Kubeclient
+  module Common
+    # Wrapper around Net:HTTP to support k8s services
+    class HttpProxyWrapper
+      def initialize(uri, options)
+        @uri = uri
+        @options = options
+      end
+
+      def get(path)
```

Comment:
How will you know when they're needed? Don't like the idea of adding them in one at a time when issues come in - should be simple enough to do in one go now I would have thought, delegating to a wrapped HTTP object?